File: app/src/main/java/com/example/cleveradsapp/AnotherBannerActivity.java
package com.example.cleveradsapp;

import androidx.annotation.RequiresApi;
import androidx.appcompat.app.AppCompatActivity;

import android.os.Build;
import android.os.Bundle;

public class AnotherBannerActivity extends AppCompatActivity {

    @RequiresApi(api = Build.VERSION_CODES.Q)
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_another_banner);
        this.registerActivityLifecycleCallbacks(ActivityHolder.getInstance());
    }
}

File: app/src/main/java/com/example/cleveradsapp/AnotherBannerActivity.kt
package com.example.cleveradsapp

import android.os.Build
import android.os.Bundle
import androidx.annotation.RequiresApi
import androidx.appcompat.app.AppCompatActivity
import com.example.cleveradsapp.ActivityHolder.Companion.instance

class AnotherBannerActivity : AppCompatActivity() {
    @RequiresApi(api = Build.VERSION_CODES.Q)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_another_banner)
        registerActivityLifecycleCallbacks(instance!!)
    }
}